### PR TITLE
fix: `react` projects with `eslint` are broken

### PR DESCRIPTION
**Pull request by iliapolo**
The following [PR](https://github.com/projen/projen/pull/1281) bumped `eslint` major version to `8.0` to address a security vulnerability.

However, the current major version of `react-scripts` that is configured by projen (4.0) is not compatible with it. As observed in an attempt to upgrade dependencies in the [construct-hub-webapp](https://github.com/cdklabs/construct-hub-webapp/runs/4632426462?check_suite_focus=true) project:

```console
build » compile | react-scripts build

There might be a problem with the project dependency tree.
It is likely not a bug in Create React App, but something you need to fix locally.

The react-scripts package provided by Create React App requires a dependency:

  "eslint": "^7.11.0"

Don't try to install it manually: your package manager does it automatically.
However, a different version of eslint was detected higher up in the tree:

  /home/runner/work/construct-hub-webapp/construct-hub-webapp/node_modules/eslint (version: 8.5.0) 
```

There is a new major version of [`react-scripts`](https://github.com/facebook/create-react-app/tree/v5.0.0/packages/react-scripts) that is compatible with eslint 8.0.

This will also have the benefit of unblocking the dependency chain upgrade for resolving a `browserlist` [CVE](https://github.com/cdklabs/construct-hub-webapp/security/dependabot/yarn.lock/browserslist/open), which is currently stuck on old major versions of `react-script` (and its dependencies). 

Note that version `5.0.0` does come with some [breaking changes](https://github.com/facebook/create-react-app/releases/tag/v5.0.0). Most notably, dropping support for node 10 and node 12. I think we can probably afford this though.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/web/react.ts [hunk 1]
```diff
@@ -182,7 +183,13 @@ export class ReactComponent extends Component {
...
     project.compileTask.reset();
 
-    project.addDeps('react', 'react-dom', 'react-scripts@^4.0.0', 'web-vitals');
+    // validate min node version as required by `react-scripts@5`.
+    // https://github.com/facebook/create-react-app/blob/v5.0.0/packages/react-scripts/package.json#L12
+    if (project.package.minNodeVersion && semver.gt('14.0.0', project.package.minNodeVersion)) {
+      throw new Error(`React projects require a minimun node version of 14.0.0 (got ${project.package.minNodeVersion})`);
+    }
+
+    project.addDeps('react', 'react-dom', 'react-scripts@^5.0.0', 'web-vitals');
```
Comment (eladb) [inline src/web/react.ts:192]: nit:

```suggestion
    project.addDeps(
      'react', 
      'react-dom',
      'react-scripts@^5',
      'web-vitals');
```

**Review by eladb — CHANGES_REQUESTED** [review]

File: src/web/react.ts [hunk 2]
```diff
@@ -182,7 +183,13 @@ export class ReactComponent extends Component {
     // No compile for react app
     project.compileTask.reset();
 
-    project.addDeps('react', 'react-dom', 'react-scripts@^4.0.0', 'web-vitals');
+    // validate min node version as required by `react-scripts@5`.
+    // https://github.com/facebook/create-react-app/blob/v5.0.0/packages/react-scripts/package.json#L12
+    if (project.package.minNodeVersion && semver.gt('14.0.0', project.package.minNodeVersion)) {
```
[the remark this answers is not in the document]
Comment (iliapolo) [reply src/web/react.ts:188]: Fair enough

**Review by eladb — APPROVED** [review]

**Comment by iliapolo** [comment]
@eladb Any idea why this isn't merging?

**Comment by eladb** [comment]
No, but would be amazing to finally move from mergify to GitHub workflows

**Comment by iliapolo** [comment]
> No, but would be amazing to finally move from mergify to GitHub workflows

Yeap, but this time its actually working fine :)

What happened is that the self-mutation step cancelled the workflow, which cancelled the `integ` job (among others) and set the `build` job to success. But this isn't enough for merging because `integ` is also a required check. 

Not sure what to do here of the top of my head, but we need to address this.

Maybe we need to forgo the status check updates and workflow cancellation, and just trigger the workflow again on the self-mutation commit from the workflow, wonder if thats possible.

**Comment by eladb** [comment]
> Maybe we need to forgo the status check updates and workflow cancellation, and just trigger the workflow again on the self-mutation commit from the workflow, wonder if thats possible.

Yes I was wondering the same thing. Would be much more elegant.

**Pull request merged**